s3/replica_client.go

```diff
@@ -525,7 +551,27 @@ func (itr *fileIterator) Next() bool {
 
 			// Set file info
 			info.Size = aws.ToInt64(obj.Size)
-			info.CreatedAt = aws.ToTime(obj.LastModified)
+
+			// Use fast LastModified timestamp by default
+			createdAt := aws.ToTime(obj.LastModified).UTC()
+
+			// Only fetch accurate timestamp from metadata when requested (timestamp-based restore)
+			if !itr.timestamp.IsZero() {
+				head, err := itr.client.s3.HeadObject(itr.ctx, &s3.HeadObjectInput{
+					Bucket: aws.String(itr.client.Bucket),
+					Key:    obj.Key,
+				})
+
+				if err == nil && head.Metadata != nil {
+					if ts, ok := head.Metadata[litestream.MetadataKeyTimestamp]; ok {
+						if parsed, err := time.Parse(time.RFC3339Nano, ts); err == nil {
+							createdAt = parsed
+						}
+					}
+				}
```

Comment:
Return error if one occurs. We don't want to fallback to AWS last modified time if the `HEAD` fails for some reason.